xgcm/test/test_transform.py

```diff
@@ -0,0 +1,723 @@
+    target_data = transform_kwargs.setdefault("target_data", None)
+
+    if transform_kwargs["target_data"] is None:
+        # When no target_data is provided default to axis coordinates depending on method
+        if transform_kwargs["method"] == "linear":
+            expected_data_coord = grid.axes[axis].coords["center"]
+        elif transform_kwargs["method"] == "conservative":
+            expected_data_coord = grid.axes[axis].coords["outer"]
+    else:
+        # When target_data is provided check against the name of the dataarray
+        expected_data_coord = target_data.name
+
+    target = target.data
+
+    if error_flag:
+        with pytest.xfail():
+            transformed = grid.transform(source.data, axis, target, **transform_kwargs)
+    else:
+        transformed = grid.transform(source.data, axis, target, **transform_kwargs)
+        assert expected_data_coord in transformed.coords
+
+
+""" Multidimensional tests with dask scheduler """
+
+
+# dask client fixtures (adapted from https://github.com/xgcm/fastjmd95/blob/2cbccf2144e73c13a91741029eb7fd273d380d28/fastjmd95/test/test_jmd95.py#L61)
+@pytest.fixture
+def no_client():
+    return None
+
+
+@pytest.fixture
+def threaded_client():
+    with dask.config.set(scheduler="threads"):
+        yield
+
+
+@pytest.fixture
+def processes_client():
+    with dask.config.set(scheduler="processes"):
+        yield
+
+
+@pytest.fixture(scope="module")
+def distributed_client():
+    from dask.distributed import Client, LocalCluster
+
+    cluster = LocalCluster(threads_per_worker=1, n_workers=2, processes=True)
+    client = Client(cluster)
+    yield
+    client.close()
+    del client
+    cluster.close()
+    del cluster
+
+
+all_clients = ["no_client", "threaded_client", "processes_client", "distributed_client"]
+
+
+@pytest.mark.parametrize("client", all_clients)
+def test_grid_transform_multidim(request, client, multidim_cases):
+    # broadcast the 1d column agains some other dims and make sure that the 1d results are still valid
+    source, grid_kwargs, target, transform_kwargs, expected, error_flag = multidim_cases
+
+    na = 3
+    source = source * xr.DataArray(np.ones([na]), dims=["a"])
+
+    if client != "no_client":
+        source = source.chunk({"a": 1})
+
+    # calculate the multidimensional result
+    axis = list(grid_kwargs["coords"].keys())[0]
+    grid = Grid(source, **grid_kwargs)
+
+    # compare for each depth column
+    if error_flag:
+        with pytest.xfail():
+            transformed = grid.transform(source.data, axis, target, **transform_kwargs)
+    else:
+        client = request.getfixturevalue(client)
+        transformed = grid.transform(
+            source.data, axis, target, **transform_kwargs
+        ).compute(scheduler=client)
+        for aa in range(na):
+            transformed_column = transformed.isel({"a": aa})
+            xr.testing.assert_allclose(transformed_column, expected.data)
```

Comment:
Let's not use a loop here. Instead broadcast `expected` against `transformed`.

Also, why are we not doing `xr.testing.assert_allclose(transformed_column, expected)`? What do we need `.data`?

Reply:
Same issue has appeared before. `transformed` is a dataarray, `expected` a dataset. I do that to futureproof the test architecture. Eventually I want to be able to output full datasets when the input is a dataset.

Reply:
> Let's not use a loop here. Instead broadcast expected against transformed.

Sounds good. Ill implement after lunch.

Reply:
Are you ok with leaving the datasets in there? I think it will make the setup of the next PR a lot easier at the cost of a bit less readability now.